small-talk/app/accountCreate/page.test.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { fireEvent, render, waitFor, screen } from '@testing-library/react';
+import RegisterPage from './page.js'; // Ensure this path is correct
+import userExists from "@/db/username";
+import handleRegister from "@/db/registerToDB";
+import validateInput from '@/components/inputValidation';
+import vulgar from '@/components/vulgarLang';
+
+jest.mock("@/db/username");
+jest.mock("@/db/registerToDB");
+jest.mock('@/components/inputValidation');
+jest.mock('@/components/vulgarLang');
+
+describe('RegisterPage Component', () => {
+
+    beforeEach(() => {
+        // Mock window.alert
+        window.alert = jest.fn();
+      });
+      
+  it('renders correctly', () => {
+    render(<RegisterPage />);
+    screen.debug(); // This will print the rendered output to the console
+  });
+
+  it('allows input fields to be filled', () => {
+    render(<RegisterPage />);
+    const firstNameInput = screen.getByPlaceholderText('John');
+    const lastNameInput = screen.getByPlaceholderText('Doe');
+    const dobInput = screen.getByPlaceholderText('Date of Birth');
+    const usernameInput = screen.getByPlaceholderText('Username');
+    const passwordInput = screen.getByPlaceholderText('Password');
+
+    fireEvent.change(firstNameInput, { target: { value: 'Test' } });
+    fireEvent.change(lastNameInput, { target: { value: 'User' } });
+    fireEvent.change(dobInput, { target: { value: '1990-01-01' } });
+    fireEvent.change(usernameInput, { target: { value: 'testuser' } });
+    fireEvent.change(passwordInput, { target: { value: 'Test@1234' } });
+
+    expect(firstNameInput.value).toBe('Test');
+    expect(lastNameInput.value).toBe('User');
+    expect(dobInput.value).toBe('1990-01-01');
+    expect(usernameInput.value).toBe('testuser');
+    expect(passwordInput.value).toBe('Test@1234');
```

Comment:
I'm not too familiar with how this performs the checks for expected values. Does this test go through as expected? Or does this need to take the hash function into consideration because when the password gets sent to the db, the hashed password is stored, not the plaintext.

Reply:
I believe those checks just check that the input written to the fields match what's actually there on the page. So before the account is actually created.